github/search.go

```diff
@@ -15,6 +15,13 @@ import (
 // SearchService provides access to the search related functions
 // in the GitHub API.
 //
+// Each method takes a query string defining the search keywords and any search qualifiers. For example,
+// when searching issues, the query "gopher is:issue language:go" will search for issues containing the word "gopher"
+// in go repositories. The method call
+//   opts :=  &github.SearchOptions{Sort: "created", Order: "asc"}
+//   cl.Search.Issues(ctx, "gopher is:issue label:bug language:go", opts)
+// will search for such issues, sorting by creation date in ascending order (i.e., oldest first).
+//
```

Comment:
I understand the idea of placing this doc section under `SearchService`, it applies to all its methods. But I worry it might be harder to see/find if it's here.

It might be better to just move it to the `Issues` or `Repositories` methods. I feel people are more likely to spot it then.

No need to have it everyone, just the most prominent method is fine.

What do you think?

Reply:
It's true that searching issues is the most common use case, but I don't see how `Issues` is the most prominent method from either the godoc or the source code itself, as it is in the middle of the module's methods. An engineer will still have to scan all the methods of `SearchService` for the example section if they start searching by code or repositories. 

It might make more sense to me if we could choose the order of methods in godoc, but the godoc alphabetical ordering makes it hard to mark a particular method as more important.

Reply:
They show up almost entirely [on one page](https://user-images.githubusercontent.com/1924134/30095214-2d004e78-929f-11e7-9e13-42f6c8676b6b.png) for me.

All right, it's good as is. We can't guarantee everyone will see it. We just want to have it somewhere so that the people who do look can find it.